tensorflow/core/kernels/segment_reduction_ops.cc

```diff
@@ -332,7 +330,8 @@ class UnsortedSegmentSumOp : public UnsortedSegmentBaseOp<Device, T, Index> {
   explicit UnsortedSegmentSumOp(OpKernelConstruction* context)
       : UnsortedSegmentBaseOp<Device, T, Index>(
             context,
-           *(new functor::UnsortedSegmentSumFunctor<Device, T, Index>())) {}
+            sumFunctor) {}
+    functor::UnsortedSegmentSumFunctor<Device, T, Index> sumFunctor;
```

Comment:
I don't think that the constructors are called in the right order. I now see why you had it this way, but that's not good.

You could fix that construction part by using a pointer, then passing it, but the destruction cannot be fixed because it's the wrong order.

There are ways to fix it, but in general, that is usually just a symptom that the design could probably be improved. Could you re-think that part? I'm happy to chat on another channel if you need input.